xgcm/autogenerate.py

```diff
@@ -0,0 +1,312 @@
+        #                                 boundary=boundary,
+        #                                 fill_value=fill_value)
+        # ax__neighbor_binary_func_raw(da, ax., to,
+        #                                           boundary=boundary,
+        #                                           fill_value=fill_value)
+
+        # Place the correct attributes
+        ds[name] = fill_attrs(ds[name], pos_from, axis)
+        ds[new_name] = fill_attrs(ds[new_name], pos_to, axis)
+    else:
+        ax = Axis(ds, axis, periodic=periodic, wrap=wrap)
+        ds.coords[new_name] = ax.interp(ds[name], pos_to, boundary=boundary,
+                                        fill_value=fill_value)
+    return ds
+
+
+def generate_grid_ds(ds,
+                     axes_dims_dict,
+                     axes_coords_dict=None,
+                     position=None,
+                     wrap=None,
+                     pad=None,
+                     new_name=None,
+                     generate_distance=True,
+                     distance_name=None,
+                     ll_dist=True):
+    """
+    Add c-grid dimensions and coordinates (optional) to observational Dataset
+
+    Parameters
+    ----------
+    ds : xarray.Dataset
+     Dataset with gridinformation used to construct c-grid
+    axes_dims_dict : dict
+     Dict with information on the dimension in ds corrsponding to the xgcm
+     axis. E.g. {'X':'lon','Y':'lat'}
+    axes_coords_dict : dict
+     Dict with information on the coordinates in ds corrsponding to the
+     xgcm axis. E.g. {'X':'geolon','Y':'geolat'}
+    position : {None,tuple, dict}
+     Position of the gridpoints given in 'ds' and the desired position to be
+     generated. Defaults to ('center','left'). Can be a tuple like
+     ('center','left'), or a dict with corresponding axes
+     (e.g. {'X':('center','left'),'Z':('left','center')})
+    wrap : {None, float, dict}
+     Specifies the discontinuity at the boundary to wrap e.g. longitudes
+     without artifacts. Can be defined globally (for all fields defined in
+     axes_dims_dict and axes_coords_dict) {float, None} or per dataset
+     variable (dict e.g. {'longitude':360,'latitude':180})
+    pad : {None, float, 'auto'}
+     Specifies the discontinuity at the boundary to wrap e.g. longitudes
+     without artifacts. Can be defined globally (for all fields defined in
+     axes_dims_dict and axes_coords_dict) {float, None} or per dataset
+     variable ({dict} e.g. {'z':'auto','latitude':0.0})
+    new_name : str
+     Name of the inferred grid variable. Defaults to name+'_inferred'
+    generate_distance : Bool
+     Switch for the generation of distances based on the axes_coords_dict
+    distance_name : str
+     Name prefix for distance coordinates. Defaults to 'd'+name.
+    ll_dist : Bool
+     Activates the conversion of lon/lat coordinates to distances in meters.
+     Requires coordinates to be specified on 'X' and 'Y'.
```

Comment:
It doesn't look like this option is ever utilized.

I'm confused about what it would mean. How can you covert lon/lat coordinates to distance without specifying a projection?